test/common_utils.py

```diff
@@ -202,3 +202,47 @@ def _test_fn_on_batch(batch_tensors, fn, scripted_fn_atol=1e-8, **fn_kwargs):
         # scriptable function test
         s_transformed_batch = scripted_fn(batch_tensors, **fn_kwargs)
         torch.testing.assert_close(transformed_batch, s_transformed_batch, rtol=1e-5, atol=scripted_fn_atol)
+
+
+def cache(fn):
+    """Similar to :func:`functools.cache` (Python >= 3.8) or :func:`functools.lru_cache` with infinite buffer size,
+    but also caches exceptions.
+
+    .. warning::
+
+        Only use this on deterministic functions.
+    """
+    sentinel = object()
+    out_cache = {}
+    exc_cache = {}
+
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
+        key = args + tuple(kwargs.values())
+
+        out = out_cache.get(key, sentinel)
+        if out is not sentinel:
+            return out
+
+        exc = exc_cache.get(key, sentinel)
+        if exc is not sentinel:
+            raise exc
+
+        try:
+            out = fn(*args, **kwargs)
+        except Exception as exc:
+            exc_cache[key] = exc
+            raise exc
+
+        out_cache[key] = out
+        return out
+
+    return wrapper
+
+
+@cache
+def script(fn):
+    try:
+        return torch.jit.script(fn)
+    except Exception as error:
+        raise AssertionError(f"Trying to `torch.jit.script` '{fn.__name__}' raised the error above.") from error
```

Comment:
@pmeier do we really need to cache scripted functions ?

Reply:
See https://github.com/pytorch/vision/pull/6606#discussion_r974211565. Let's re-evaluate if we can actually test things rather than xfailing in the first step.